File: easy/height_checker.go
package easy

import "sort"

func HeightChecker(heights []int) int {
	count := 0
	arr := make([]int, len(heights))
	copy(arr[:], heights)
	sort.Ints(arr)

	for i := range heights {
		if heights[i] != arr[i] {
			count++
		}
	}
	return count
}

File: src/easy/height-checker.ts
export const heightChecker = (heights: number[]): number => {
    const nums = [...heights];
    nums.sort((a, b) => a - b);
    console.log(nums, heights);
    return nums.reduce((a, b, i) => a + (heights[i] !== b ? 1 : 0), 0);
};
